hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/StreamStageTest.java

```diff
@@ -898,6 +905,30 @@ public void merge() {
                 streamToString(sinkStreamOf(Integer.class), formatFn));
     }
 
+    @Test
+    public void when_mergeTimestampedToNonTimestamped_then_error() {
+        StreamStage<SimpleEvent> timestamped = p.readFrom(TestSources.itemStream(1)).withIngestionTimestamps();
+        StreamStage<SimpleEvent> nonTimestamped = p.readFrom(TestSources.itemStream(1)).withoutTimestamps();
+
+        // Then
+        exception.expectMessage("both have or both not have timestamp definitions");
+
+        // When
+        nonTimestamped.merge(timestamped);
```

Comment:
I think this is meant to be `timestamped.merge(nonTimestamped)`

Reply:
It's disputable, test name is `mergeTimestampedToNonTimestamped`, in `A.merge(B)` I consider that I'm merging `B` into `A`. Most importantly, there are two methods and they merge both ways.

Reply:
ah, no, the other method is same 😒